mozilla/blurts-server
# MNTOR-949 - subscription platform integration for nextjs app

# References: 
Jira: MNTOR-949




# Description

This provides basic subplat integration. Currently it demonstrates:

1. add new `subscription` `Array<string>` to `session.user.fxa` object 
3. example "subscribe" button for staging subplat
4. example welcome screen and badge in user menu

This PR is based on a recent spike, and still needs the following to be truly ready for review:

- [x] move configuration to env vars
- [x] add flag (disabled by default)
- [x] add example UI (also behind flag)

Per the [next-auth docs](https://next-auth.js.org/getting-started/client#updating-the-session), we should be able to update the session without requiring the user to log in again.

# How to test

1. sign in using unsubscribed account
2. click "Subscribe" button on user dashboard page
3. complete purchase using [stripe test card](https://stripe.com/docs/testing)
5. ensure that:
 a. redirected to the dashboard after login
 b. "Subscribe" button on user dashboard does not appear
6. Navigate to Firefox account settings and unsubscribe
7. Log out/in and ensure that premium features are still available for 1 month
8. TODO figure out how to test that cancellation worked without having to wait 1 month

# Checklist (Definition of Done)
